mis_builder/models/mis_builder.py

```diff
@@ -211,9 +212,8 @@ def _render_num(self, lang_id, value, divider,
             '%%%s.%df' % (sign, dp),
             value,
             grouping=True)
-        value = u'%s\N{NO-BREAK SPACE}%s%s' % \
-            (value, divider_label, suffix or '')
-        value = value.replace('-', u'\N{NON-BREAKING HYPHEN}')
```

Comment:
Please keep this line replacing '-' by non-breaking hyphen, otherwise you may get ugly result with the negative sign not on the same line as the number.